### PR TITLE
Delete operation portal

### PR DESCRIPTION
**Pull request by MaazMS**
create delete method of portal 

* create log of portal 
```bash
INFO	controllers.Workshop	Adding Finalizer for the Workshop	{"workshop": "workshop-infra/cloud-native-workshop"}
time="2021-10-10T07:50:40Z" level=info msg="Creating Redis" source="portal_reconciler.go:54"
time="2021-10-10T07:50:40Z" level=info msg="Created redis Secret" source="portal_reconciler.go:59"
time="2021-10-10T07:50:40Z" level=info msg="Created redis Persistent Volume Claim" source="portal_reconciler.go:66"
time="2021-10-10T07:50:41Z" level=info msg="Created redis Deployment" source="portal_reconciler.go:74"
time="2021-10-10T07:50:41Z" level=info msg="Created redis Service" source="portal_reconciler.go:95"
time="2021-10-10T07:50:41Z" level=info msg="Creating portal" source="portal_reconciler.go:104"
time="2021-10-10T07:50:41Z" level=info msg="Created portal Deployment" source="portal_reconciler.go:110"
time="2021-10-10T07:50:41Z" level=info msg="Created portal Service" source="portal_reconciler.go:132"
time="2021-10-10T07:50:41Z" level=info msg="Created portal Route" source="portal_reconciler.go:140"
time="2021-10-10T07:50:41Z" level=info msg="Created gitea Project" source="gitea_reconciler.go:73"
time="2021-10-10T07:50:41Z" level=info msg="Created giteas.gpte.opentlc.com Custom Resource Definition" source="gitea_reconciler.go:81"
time="2021-10-10T07:50:41Z" level=info msg="Created gitea-operator Service Account" source="gitea_reconciler.go:89"
time="2021-10-10T07:50:42Z" level=info msg="Created gitea-operator Cluster Role" source="gitea_reconciler.go:97"
time="2021-10-10T07:50:42Z" level=info msg="Created gitea-operator Cluster Role Binding" source="gitea_reconciler.go:105"
time="2021-10-10T07:50:42Z" level=info msg="Created gitea-operator Operator" source="gitea_reconciler.go:114"
I1010 07:50:43.196884       1 request.go:621] Throttling request took 1.042545221s, request: GET:https://172.30.0.1:443/apis/security.openshift.io/v1?timeout=32s
time="2021-10-10T07:50:45Z" level=info msg="Created gitea-server Custom Resource" source="gitea_reconciler.go:122"
time="2021-10-10T07:50:45Z" level=info msg="Waiting for deployment gitea-server. Default timeout: 420 seconds" source="k8s_helpers.go:68"

time="2021-10-10T07:52:40Z" level=info msg="Deployment 'gitea-server' successfully scaled to 1" source="k8s_helpers.go:81"
2021-10-10T07:52:40.589Z	DEBUG	controller	Successfully Reconciled	{"reconcilerGroup": "workshop.stakater.com", "reconcilerKind": "Workshop", "controller": "workshop", "name": "cloud-native-workshop", "namespace": "workshop-infra"}
time="2021-10-10T07:52:40Z" level=info msg="OpenShift Console URL https://console-openshift-console.apps.workshop.6hf8sw9d.kubeapp.cloud" source="workshop_controller.go:103"
time="2021-10-10T07:52:40Z" level=info msg="Apps Hostname Suffix apps.workshop.6hf8sw9d.kubeapp.cloud" source="workshop_controller.go:108"
time="2021-10-10T07:52:40Z" level=info msg="Creating Redis" source="portal_reconciler.go:54"
time="2021-10-10T07:52:40Z" level=info msg="Creating portal" source="portal_reconciler.go:104"
time="2021-10-10T07:52:40Z" level=info msg="Updated portal Deployment" source="portal_reconciler.go:122"
time="2021-10-10T07:52:41Z" level=info msg="Waiting for deployment gitea-server. Default timeout: 420 seconds" source="k8s_helpers.go:68"
time="2021-10-10T07:52:41Z" level=info msg="Deployment 'gitea-server' successfully scaled to 1" source="k8s_helpers.go:81"
2021-10-10T07:52:41.234Z	DEBUG	controller	Successfully Reconciled	{"reconcilerGroup": "workshop.stakater.com", "reconcilerKind": "Workshop", "controller": "workshop", "name": "cloud-native-workshop", "namespace": "workshop-infra"}

``` 
* delete log of portal 
```bash
INFO	controllers.Workshop	Deleting workshopcloud-native-workshop	{"workshop": "workshop-infra/cloud-native-workshop"}
time="2021-10-10T08:02:10Z" level=info msg="Deleting  portal " source="portal_reconciler.go:201"
time="2021-10-10T08:02:10Z" level=info msg="Deleted portal Route" source="portal_reconciler.go:207"
time="2021-10-10T08:02:10Z" level=info msg="Deleted portal Service" source="portal_reconciler.go:214"
time="2021-10-10T08:02:10Z" level=info msg="Deleted portal Deployment" source="portal_reconciler.go:224"
time="2021-10-10T08:02:10Z" level=info msg=" Deleted portal  successfully" source="portal_reconciler.go:226"
time="2021-10-10T08:02:10Z" level=info msg="Deleting Redis" source="portal_reconciler.go:163"
time="2021-10-10T08:02:10Z" level=info msg="Deleted redis Service" source="portal_reconciler.go:169"
time="2021-10-10T08:02:10Z" level=info msg="Deleted redis Deployment " source="portal_reconciler.go:176"
time="2021-10-10T08:02:10Z" level=info msg="Deleted redis Persistent Volume Claim" source="portal_reconciler.go:183"
time="2021-10-10T08:02:10Z" level=info msg="Deleted redis Secret" source="portal_reconciler.go:190"
time="2021-10-10T08:02:10Z" level=info msg="Deleted Redis successfully" source="portal_reconciler.go:191"
time="2021-10-10T08:02:10Z" level=info msg="Deleting gitea" source="gitea_reconciler.go:202"
time="2021-10-10T08:02:10Z" level=info msg="Deleted gitea-server gitea Custom Resource" source="gitea_reconciler.go:212"
time="2021-10-10T08:02:10Z" level=info msg="Deleted gitea-operator gitea Operator" source="gitea_reconciler.go:219"
time="2021-10-10T08:02:11Z" level=info msg="Deleted gitea-operator gitea Cluster  Role Binding" source="gitea_reconciler.go:226"
time="2021-10-10T08:02:11Z" level=info msg="Deleted gitea-operator gitea Cluster Role" source="gitea_reconciler.go:233"
time="2021-10-10T08:02:11Z" level=info msg="Deleted gitea-operator gitea Service Account" source="gitea_reconciler.go:240"
time="2021-10-10T08:02:11Z" level=info msg="Deleted giteas.gpte.opentlc.com gitea Custom Resource Definition" source="gitea_reconciler.go:247"
time="2021-10-10T08:02:11Z" level=info msg="Deleted gitea gitea Project " source="gitea_reconciler.go:254"
time="2021-10-10T08:02:11Z" level=info msg="Gitea deleted succesfully" source="gitea_reconciler.go:255"
time="2021-10-10T08:02:11Z" level=info msg="Deleting nexus" source="nexus_reconciler.go:128"
I1010 08:02:12.376856       1 request.go:621] Throttling request took 1.047166077s, request: GET:https://172.30.0.1:443/apis/project.openshift.io/v1?timeout=32s
time="2021-10-10T08:02:18Z" level=info msg="Finalizer removed for workshopcloud-native-workshop" source="workshop_controller.go:131"
2021-10-10T08:02:19.007Z	DEBUG	controller	Successfully Reconciled	{"reconcilerGroup": "workshop.stakater.com", "reconcilerKind": "Workshop", "controller": "workshop", "name": "cloud-native-workshop", "namespace": "workshop-infra"}
time="2021-10-10T08:02:19Z" level=info msg="Workshop resource not found. Ignoring since object must be deleted" source="workshop_controller.go:81"
2021-10-10T08:02:19.007Z	DEBUG	controller	Successfully Reconciled	{"reconcilerGroup": "workshop.stakater.com", "reconcilerKind": "Workshop", "controller": "workshop", "name": "cloud-native-workshop", "namespace": "workshop-infra"}

```

### CODE REVIEW

**Comment by stakater-user** [comment]
@MaazMS Image is available for testing. `docker pull stakater/workshop-operator:SNAPSHOT-PR-32-6a449936`

File: controllers/portal_reconciler.go [hunk 1]
```diff
@@ -208,37 +183,22 @@ func (r *WorkshopReconciler) deleteRedis(workshop *workshopv1.Workshop) (reconci
...
-	route := kubernetes.NewSecuredRoute(workshop, r.Scheme, serviceName, workshop.Namespace, labels, serviceName, 8080)
-	routeFound := &routev1.Route{}
-	routeErr := r.Get(context.TODO(), types.NamespacedName{Name: route.Name, Namespace: workshop.Namespace}, routeFound)
-	if routeErr == nil {
-		// Delete Route
-		if err := r.Delete(context.TODO(), route); err != nil {
-			return reconcile.Result{}, err
-		}
-		log.Infof("Deleted %s Route", route.Name)
+	log.Info("Deleting gitea UpdateUsernameDistribution ")
```
Comment (rasheedamir) [inline controllers/portal_reconciler.go]: @MaazMS `gitea`?

File: controllers/workshop_controller.go [hunk 2]
```diff
@@ -246,10 +219,17 @@ func (r *WorkshopReconciler) SetupWithManager(mgr ctrl.Manager) error {
...
+func (r *WorkshopReconciler) handleDelete(ctx context.Context, req ctrl.Request, workshop *workshopv1.Workshop, userID int, appsHostnameSuffix string, openshiftConsoleURL string) (ctrl.Result, error) {
 	log := r.Log.WithValues("workshop", req.NamespacedName)
 	log.Info("Deleting workshop" + workshop.ObjectMeta.Name)
 
+	if result, err := r.deleteUpdateUsernameDistribution(workshop, userID, appsHostnameSuffix, openshiftConsoleURL); util.IsRequeued(result, err) {
+		return result, err
+	}
+	if result, err := r.deleteRedis(workshop); util.IsRequeued(result, err) {
+		return result, err
+	}
```
Comment (rasheedamir) [inline controllers/workshop_controller.go:252]: @MaazMS write one method deletePortal and put these two methods in it

File: controllers/portal_reconciler.go (hunk 1 above)
Comment (MaazMS) [reply controllers/portal_reconciler.go]: i will remove gitea .

File: controllers/workshop_controller.go (hunk 2 above)
Comment (MaazMS) [reply controllers/workshop_controller.go:252]: yes, i will create single method of deletePortal and insdie deletePortal i will create two method

File: controllers/portal_reconciler.go [hunk 3]
```diff
@@ -17,6 +17,23 @@ import (
 	"github.com/stakater/workshop-operator/common/util"
 )
 
+const (
+	REDISPERSISTENTVOLUMECLAIM = "512Mi"
```
Comment (rasheedamir) [inline controllers/portal_reconciler.go]: @MaazMS fix these names as well

Comment (MaazMS) [reply controllers/portal_reconciler.go]: ok, i will fix const name.

**Comment by stakater-user** [comment]
@MaazMS Image is available for testing. `docker pull stakater/workshop-operator:SNAPSHOT-PR-32-7c9fdafc`

**Comment by stakater-user** [comment]
@MaazMS Image is available for testing. `docker pull stakater/workshop-operator:SNAPSHOT-PR-32-4b2055d1`

File: controllers/portal_reconciler.go [hunk 4]
```diff
@@ -209,36 +198,22 @@ func (r *WorkshopReconciler) deleteRedis(workshop *workshopv1.Workshop) (reconci
 func (r *WorkshopReconciler) deleteUpdateUsernameDistribution(workshop *workshopv1.Workshop,
```
Comment (tanalam2411) [inline controllers/portal_reconciler.go]: @MaazMS Rename this method `deleteUpdateUsernameDistribution` to `deleteUsernameDistribution`

File: config/samples/workshop_v1_cloud_native_workshop.yaml [hunk 5]
```diff
@@ -13,7 +13,7 @@ spec:
   infrastructure:
     guide:
       scholars:
-        enabled: false
+        enabled: true
```
Comment (tanalam2411) [inline config/samples/workshop_v1_cloud_native_workshop.yaml]: @MaazMS  I think by default `enabled` should be `false` for all resources in sample yaml. And who ever is using the sample yaml can enable it for testing but should not commit it.

**Comment by stakater-user** [comment]
@MaazMS Image is available for testing. `docker pull stakater/workshop-operator:SNAPSHOT-PR-32-79373175`

**Review by tanalam2411 — CHANGES_REQUESTED** [review]

File: config/samples/workshop_v1_cloud_native_workshop.yaml (hunk 5 above)
Comment (MaazMS) [reply config/samples/workshop_v1_cloud_native_workshop.yaml]: @tanalam2411  i will do `enabled : false`   for all resources in sample yaml.

**Comment by stakater-user** [comment]
@MaazMS Image is available for testing. `docker pull stakater/workshop-operator:SNAPSHOT-PR-32-89012b30`

**Comment by rasheedamir** [comment]
@MaazMS resolve conflicts

**Comment by stakater-user** [comment]
@MaazMS Image is available for testing. `docker pull stakater/workshop-operator:SNAPSHOT-PR-32-58a146b1`

File: controllers/portal_reconciler.go [hunk 6]
```diff
@@ -17,6 +17,23 @@ import (
 	"github.com/stakater/workshop-operator/common/util"
 )
 
+const (
+	REDIS_PERSISTENT_VOLUMECLAIM = "512Mi"
```
Comment (rasheedamir) [inline controllers/portal_reconciler.go]: shouldn't it be _CLAIM?

Comment (MaazMS) [reply controllers/portal_reconciler.go]: @rasheedamir  i will change variable `REDIS_PERSISTENT_VOLUMECLAIM` to `REDIS_PERSISTENT_VOLUME_CLAIM`

**Comment by stakater-user** [comment]
@MaazMS Image is available for testing. `docker pull stakater/workshop-operator:SNAPSHOT-PR-32-2d0c4969`

**Comment by stakater-user** [comment]
@MaazMS Image is available for testing. `docker pull stakater/workshop-operator:SNAPSHOT-PR-32-0d997eb9`

**Review by tanalam2411 — DISMISSED** [review]
LGTM

File: controllers/portal_reconciler.go [hunk 7]
```diff
@@ -17,6 +17,24 @@ import (
 	"github.com/stakater/workshop-operator/common/util"
 )
 
+const (
+	REDIS_PERSISTENT_VOLUMECLAIM  = "512Mi"
+	REDIS_PERSISTENT_VOLUME_CLAIM = "512Mi"
```
Comment (rasheedamir) [inline controllers/portal_reconciler.go]: @MaazMS defined two but why?

**Comment by stakater-user** [comment]
@MaazMS Image is available for testing. `docker pull stakater/workshop-operator:SNAPSHOT-PR-32-1a24fd5d`

**Comment by stakater-user** [comment]
@MaazMS Image is available for testing. `docker pull stakater/workshop-operator:SNAPSHOT-PR-32-c99073dc`

File: controllers/portal_reconciler.go (hunk 7 above)
Comment (MaazMS) [reply controllers/portal_reconciler.go]: @rasheedamir I have changed variable REDIS_PERSISTENT_VOLUMECLAIM to variable REDIS_PERSISTENT_VOLUME_CLAIM and forget to delete  variable REDIS_PERSISTENT_VOLUMECLAIM.

**Review by rasheedamir — APPROVED** [review]

**Pull request merged**